msal/application.py

```diff
@@ -59,6 +62,16 @@ def _get_new_correlation_id():
     return str(uuid.uuid4())
 
 
+def _get_open_port():
+    import socket
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.bind(("", 0))
+    s.listen(1)
```

Comment:
There are various implementations out there. Some of them do not contain this line. Any investigation on whether that is necessary or not?

Reply:
@abhidnya13 : I've adjusted some of the goals in this PR's description, to align with what we discussed last week. All those extra features you identified can be a good-to-have in near future, but they just do not have/need to be mixed up all together. Let's focus on the topic(s) fundamental to interactive auth, and get them done.

Reply:
I agree that we should have a fundamental interactive auth PR and that should be this one. I still think PKCE should be included here too instead of taking it up later, I agree though having it in a different PR makes more sense. And the last part about customizing such a behavior is good-to-have in near future. That near future can be soon if we agree on what we want to include as I have spent enough time looking into the possibility of customizing, I do not want to leave it here. I agree though that, it should not be a part of this PR and should not block us from getting a fundamental interactive auth up and running.

Reply:
s.listen() will make the socket open for accepting connections. 1 is the value of `backlog`. So, s.listen(1) means that it is accepting one simultaneous connection. This part of code is inspired from [here](https://stackoverflow.com/questions/2838244/get-open-tcp-port-in-python/2838309#2838309).